staging/src/k8s.io/apiserver/pkg/endpoints/discovery/aggregated/wrapper.go

```diff
@@ -28,18 +28,31 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 
 	"k8s.io/apiserver/pkg/endpoints/handlers/negotiation"
+	genericfeatures "k8s.io/apiserver/pkg/features"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
 )
 
 type WrappedHandler struct {
-	s          runtime.NegotiatedSerializer
-	handler    http.Handler
-	aggHandler http.Handler
+	s              runtime.NegotiatedSerializer
+	handler        http.Handler
+	aggHandler     http.Handler
+	peerAggHandler http.Handler
 }
 
 func (wrapped *WrappedHandler) ServeHTTP(resp http.ResponseWriter, req *http.Request) {
-	mediaType, _ := negotiation.NegotiateMediaTypeOptions(req.Header.Get("Accept"), wrapped.s.SupportedMediaTypes(), DiscoveryEndpointRestrictions)
 	// mediaType.Convert looks at the request accept headers and is used to control whether the discovery document will be aggregated.
+	mediaType, _ := negotiation.NegotiateMediaTypeOptions(req.Header.Get("Accept"), wrapped.s.SupportedMediaTypes(), DiscoveryEndpointRestrictions)
 	if IsAggregatedDiscoveryGVK(mediaType.Convert) {
+		if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.UnknownVersionInteroperabilityProxy) && mediaType.Convert.Version == "v2" {
+			if mediaType.Profile == "local" {
+				LocalDiscoveryRequestCounter.Inc()
+				wrapped.aggHandler.ServeHTTP(resp, req)
+				return
+			}
+			// Serve peer-aggregated discovery by default.
+			wrapped.peerAggHandler.ServeHTTP(resp, req)
+			return
```

Comment:
allow peerAggHandler to be nil and fallback to aggHandler, so generic API servers can propagate their peer handler if they want and this will do something sensible if not set (and so kube-apiserver can just leave peerAggHandler nil if it doesn't have a peer proxy)

```suggestion
			if wrapped.peerAggHandler != nil {
				// Serve peer-aggregated discovery by default if provided.
				wrapped.peerAggHandler.ServeHTTP(resp, req)
				return
			}
```